.pre-commit-config.yaml

```diff
@@ -14,7 +27,16 @@ repos:
       - id: sourcery
         args: [--diff=git diff HEAD, --in-place, --no-summary]
 
-#  additional hooks found with in the pre-commit lib
+  # python linter
+  - repo: https://github.com/charliermarsh/ruff-pre-commit
+    rev: 'v0.0.245'
+    hooks:
+      - id: ruff
+
+  #  additional hooks found with in the pre-commit lib
+  # -- remove trailing spaces
```

Comment:
Include trailing spaces? Is that what you are referring  too? According to [PEP8](https://peps.python.org/pep-0008/#other-recommendations) Trailing white spaces is a big no no.

For mix line ending spaces, python does not like that [either](https://peps.python.org/pep-0008/#tabs-or-spaces)

Reply:
these lines are commented out, so it means that they are not being used?

Reply:
Yes they're not being used. Just used the comments for documentations purposes.

Reply:
hmm, ok - will they ever be used? If not, then you should probably remove to ensure a clean and readable config.

Reply:
The inline comments allows users to know what formatting tools are used in the `pre-commit`, technically it will be used if someone would like to know what formatting tools are used for formatting. 

I left them there because I thought the names of the programs were not self explanatory. In fact, would it be best if I just include it in the readme?

Reply:
but they're not currently used, right? They are commented out. That is my question. If they are not being used, then you should probably delete them.

(friendly reminder: you are the lead engineer on this project, you get final say)